nautobot/core/jobs/cleanup.py

```diff
@@ -48,6 +49,24 @@ class Meta:
         description = "Delete ObjectChange and/or JobResult/JobLogEntry records older than a specified cutoff."
         has_sensitive_variables = False
 
+    def recursive_delete_with_cascade(self, queryset):
+        """
+        Recursively deletes all related objects with CASCADE for a given queryset.
+
+        Args:
+            queryset (QuerySet): The queryset of objects to delete.
+        """
+        related_objects = queryset.model._meta.related_objects
+
+        for related_object in related_objects:
+            if related_object.on_delete == CASCADE:
+                related_model = related_object.related_model
+                related_field_name = related_object.field.name
+                cascade_queryset = related_model.objects.filter(**{f"{related_field_name}__in": queryset})
+                self.recursive_delete_with_cascade(cascade_queryset)
+        return queryset.delete()
+
+
     def run(self, *, cleanup_types, max_age=None):
```

Comment:
Currently we perform delete without `transaction.atomic`, wondering if `transaction.atomic` would be needed?

Reply:
Not that I can think of offhand - can you elaborate on your concern?

Reply:
Nope, it was just a taught